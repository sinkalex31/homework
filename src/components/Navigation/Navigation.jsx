import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import {
    Navbar,
    NavbarBrand,
    NavbarToggler,
    Collapse,
    Nav,
    Container,
    NavItem,
    NavLink,
} from 'reactstrap';

export default class Navigation extends Component {
    state = {
        isOpen: false,
    };

    toggle = () => this.setState({
        isOpen: !this.state.isOpen
    });

    render() {
        return (
            <Navbar dark fixed="top" expand="sm" className="bg-primary">
                <Container>
                    <Link className="navbar-brand" to="/">Customers</Link>
                    <NavbarToggler onClick={this.toggle} />
                    <Collapse isOpen={this.state.isOpen} navbar>
                        <Nav className="ml-auto" navbar>
                            <NavItem>
                                <Link className="nav-link" to="/create">Add new</Link>
                            </NavItem>
                        </Nav>
                    </Collapse>
                </Container>
            </Navbar>
        );
    }
}