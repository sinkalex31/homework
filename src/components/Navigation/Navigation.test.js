import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import Navigation from './Navigation';

describe('Navigation', () => {
    it ('should render correctly', () => {
        const tree = renderer.create(
            <MemoryRouter>
                <Navigation />
            </MemoryRouter>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});