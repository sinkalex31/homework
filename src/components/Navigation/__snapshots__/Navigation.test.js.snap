// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render correctly 1`] = `
<nav
  className="bg-primary navbar navbar-expand-sm navbar-dark fixed-top"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Customers
    </a>
    <button
      className="navbar-toggler"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      style={Object {}}
    >
      <ul
        className="ml-auto navbar-nav"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/create"
            onClick={[Function]}
          >
            Add new
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
